props/saber_fett263_buttons.h

```diff
@@ -293,248 +659,3857 @@ class SaberFett263Buttons : public virtual PropBase {
+            }
+          }
+          if (steps < 0) {
+            if (hsl_.L > 0.01) {
+              hsl_.L = clamp(hsl_.L - 0.01, 0.01, 0.5);
+              if (hsl_.L == 0.01) {
+                PlayMenuSound("mmin.wav");
+                hsl_angle_ = fusor.angle2();
+              }
+            }
+          }
+          break;         
+        }
+        ShowColorStyle::SetColor(Color16(hsl_));
+      }
+#endif
+    if (SaberBase::IsOn()) {
+      DetectSwing();
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+      if (choreo_ && saved_choreography.clash_rec[clash_count_].stance == SavedRehearsal::STANCE_END) {
+        choreo_ = false;
+        PlayMenuSound("chreoend.wav");
+      }
+#endif
+      if (auto_lockup_on_ &&
+          !swinging_ &&
+          fusor.swing_speed() > 120 &&
+          millis() - clash_impact_millis_ > saved_gesture_control.lockupdelay &&
+          SaberBase::Lockup()) {
+#ifdef FETT263_CLASH_STRENGTH_SOUND
+        clash_type_ = CLASH_LOCKUP_END;
+        RealClashSound();
+#else
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+#endif
+        auto_lockup_on_ = false;
+      }
+      if (auto_melt_on_ &&
+          !swinging_ &&
+          fusor.swing_speed() > 60 &&
+          millis() - clash_impact_millis_ > saved_gesture_control.lockupdelay &&
+          SaberBase::Lockup()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_melt_on_ = false;
+      }
+      // EVENT_PUSH
+      if (fabs(mss.x) < 3.0 &&
+          mss.y * mss.y + mss.z * mss.z > 100 &&
+          fusor.swing_speed() < 20 &&
+          fabs(fusor.gyro().x) < 5) {
+        if (millis() - push_begin_millis_ > saved_gesture_control.forcepushlen) {
+          // Checking for Clash at end of movement
+          clash_type_ = CLASH_CHECK;
+          push_begin_millis_ = millis();
+          clash_impact_millis_ = millis();
+        } 
+      } else {
+        push_begin_millis_ = millis();
+      }
+      
+    } else {
+      // EVENT_SWING - Swing On gesture control to allow fine tuning of speed needed to ignite
+      if (millis() - saber_off_time_ < MOTION_TIMEOUT) {
+        SaberBase::RequestMotion();
+        if (swinging_ && fusor.swing_speed() < 90) {
+          swinging_ = false;
+        }
+        if (!swinging_ && fusor.swing_speed() > saved_gesture_control.swingonspeed) {
+          swinging_ = true;
+          Event(BUTTON_NONE, EVENT_SWING);
+        }
+      }
+      // EVENT_THRUST
+      if (mss.y * mss.y + mss.z * mss.z < 16.0 &&
+          mss.x > 14  &&
+          fusor.swing_speed() < 150) {
+        if (millis() - thrust_begin_millis_ > 15) {
+          Event(BUTTON_NONE, EVENT_THRUST);
+          thrust_begin_millis_ = millis();
+        } 
+      } else {
+        thrust_begin_millis_ = millis();
+      }
+    }
+    if (menu_ || color_mode_ == CC_COLOR_LIST) {
+      DetectMenuTurn();
+    }
+    if (track_mode_ != PLAYBACK_OFF) {
```

Comment:
Why not put this if statement inside TrackPlayer() ?

Reply:
updated